### PR TITLE
Enforce Stricter Validation on Access Type Edits

### LINKED ISSUE #389: patch /admin/access route has no role level validation
The patch route just takes a list of updates and applies them to the database - you need extra validation that admin access levels cannot be modified.

3 edge cases to take care of here
- An admin applies a user update to demote themselves to a standard user - this is completely non-reversible since they lose access completely from their end.
- An admin promotes another user to admin on accident - the other account can now mess up all portal event and store data
- 2 admins trying to demote the access level of each other, who's access level should take precedence in this case? there should be logic so it's not just whoever acts faster gets to stay an admin

Since this is a very flexible role with full API usage, I would continue making manual DB edits to promote or demote admins which doesn't expose these edge cases to any board member with the acm portal account login credentials

### PR DESCRIPTION
**Pull request by nik-dange**
# Info

Closes #389 

# Description

Moving forward, we'll only allow admin edits to be done manually in the DB. This PR adds a few checks to ensure accidental self-demotions cannot occur, as well as the ability to demote admins/promote others to admin.

## Changes

- A few conditional checks for access type edits

# Type of Change

- [x] Patch (non-breaking change/bugfix)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to not work as
      expected)
- [ ] Documentation (A change to a README/description)
- [ ] Continuous Integration/DevOps Change (Related to deployment steps, continuous integration
      workflows, linting, etc.)
- [ ] Other: (Fill In) 

If you've selected Patch, Minor, or Major as your change type, **make sure to bump the version before merging in `package.json`!**
# Testing

I have tested that my changes fully resolve the linked issue ...

- [ ] locally.
- [ ] on the testing API/testing database.
- [ ] with appropriate Postman routes. Screenshots are included below.

# Checklist

- [ ] I have performed a self-review of my own code.
- [x] I have followed the style guidelines of this project.
- [ ] I have appropriately edited the API version in the `package.json` file.
- [ ] My changes produce no new warnings.

# Screenshots

Please include a screenshot of your Postman testing passing successfully.

### CODE REVIEW

**Comment by nik-dange** [comment]
Don't merge I didn't test yet 🥱

**Comment by nik-dange** [comment]
Screenshot 2024-01-29 at 10 13 55 PM
Screenshot 2024-01-29 at 10 14 07 PM
Screenshot 2024-01-29 at 10 14 30 PM

File: services/UserAccountService.ts [hunk 1]
```diff
@@ -225,13 +225,23 @@ export default class UserAccountService {
...
 
-      const updatedUsers = await Promise.all(accessUpdates.map(async (accessUpdate, index) => {
+      const updatedUsers = await Promise.all(accessUpdates.map(async (accessUpdate) => {
         const { user: userEmail, accessType } = accessUpdate;
 
-        const currUser = emailToUserMap[userEmail];
-        const oldAccess = currUser.accessType;
+        // Prevent a user from demoting themselves
+        if (currentUser.email === userEmail) {
+          throw new BadRequestError('Cannot alter own access level');
```
Comment (nik-dange) [inline services/UserAccountService.ts:233]: I think this should be a ForbiddenError but I'll get around to changing it in a bit

**Review by dowhep — APPROVED** [review]
LGTM 🚀 🚀 I love that you added more details to the "attempting to demote oneself" test

**Pull request merged**